### DISCUSSION TITLE
Creating Guest Network: unable to choose Network Offering. This for all L2,Shared and Isolated networks

### DISCUSSION THREAD

**Discussion by DensonNgumo** [apache/cloudstack#9716] [Q&A] [answered]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.18.0
~~~
~~~
4.19.0.1
~~~

##### CONFIGURATION

Advanced networking configured

##### OS / ENVIRONMENT

Management servers running on Ubuntu 22.0.4

##### SUMMARY

I want to create new guest networks. On the network creation window I am unable to choose any network offering. The default system network offerings as well my custom network offerings exist but on the dropdown none appear. I am thus unable to create a L2 network which will allow me to specify the VLAN.

##### STEPS TO REPRODUCE

~~~
1. Navigate to Add Network.
2.  Pick either 'Shared', 'L2' or 'Isolated'
3.  Try to specify a network offering under the dropdown provided.
~~~

##### EXPECTED RESULTS

~~~
All the available network offerings should be available for selection under the drop down.
~~~

##### ACTUAL RESULTS

No network offering is available for selection
![cloudstack_network](https://github.com/apache/cloudstack/assets/11226482/cdd54672-a740-4619-9bb5-2ad6a0bad4c0)

~~~

~~~

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by rajujith** [comment]
@DensonNgumo I see 4 network offerings for this drop-down in ACS 4.19.0.0. Can you check whether any L2 network offering is enabled?

**Comment by DensonNgumo** [comment]
Despite the default offerings, I have also created my own to test and it still does not appear.
![network-offerings](https://github.com/apache/cloudstack/assets/11226482/eb2f0316-f424-4b43-8be6-b6b9d427986d)

**Comment by weizhouapache** [comment]
> Despite the default offerings, I have also created my own to test and it still does not appear. ![network-offerings](https://private-user-images.githubusercontent.com/11226482/338532348-eb2f0316-f424-4b43-8be6-b6b9d427986d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMDI0ODUsIm5iZiI6MTcxODEwMjE4NSwicGF0aCI6Ii8xMTIyNjQ4Mi8zMzg1MzIzNDgtZWIyZjAzMTYtZjQyNC00YjQzLThiZTYtYjZiOWQ0Mjc5ODZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjExVDEwMzYyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQzYjNmOTU2ZGQxMmZiOWMyNTk1MGUxZGRjYzI1NGU3Yzg5MDQ4NGRkOGE5OGRjNWE3NWE2YjJkZTE0OGExMWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oha_ipyTG1n0g0YmZXR0eCInszU3gyqS43g99CU5lsg)

@DensonNgumo 
it works fine in my environment with 2 hypervisor types.

can you use cmk to list the network offering ?
```
cmk list networkofferings guestiptype=L2 filter=id,name zoneid=<uuid of zone>
```

**Comment by DensonNgumo** [comment]
If I specify the zone id then I am not seeing the offerings but if I don't, then I can.
![cmk-list](https://github.com/apache/cloudstack/assets/11226482/fa33ae78-0fcf-44a8-9a89-e94d37a3b438)

**Comment by DaanHoogland** [comment]
@DensonNgumo I see all offerings in both cases. and lso with or without the  guestiptype=L2. 

Did you create any offerings yourself?
Did you create this zone (and enable it) before or after creating any offerings?

You say you see this in 4.18.0 and in 4.19.0.

Did you upgrade and tested again, or create a new installation?

Can you create a new offering, with the same parameters and see if this displays?

**Comment by weizhouapache** [comment]
> If I specify the zone id then I am not seeing the offerings but if I don't, then I can. ![cmk-list](https://private-user-images.githubusercontent.com/11226482/339243280-fa33ae78-0fcf-44a8-9a89-e94d37a3b438.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgyNjU2ODEsIm5iZiI6MTcxODI2NTM4MSwicGF0aCI6Ii8xMTIyNjQ4Mi8zMzkyNDMyODAtZmEzM2FlNzgtMGZjZi00NGE4LTlhODktZTk0ZDM3YTNiNDM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEzVDA3NTYyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzM2Y0YjY0N2RmNzZjNTM0OTNiYzEyYWUzMDhhNTRjMzMwNjQ1YmViMmE4ZDYzOGQ0OGVjODhlNDg4NjFhZmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.X3zoBOa_HXzSmRjnghnXBVkEG9qnGwA1povV1CEedHI)

I suspect it is caused by the network offering "SIP_Network" created by you @DensonNgumo 
can you upload network_offerings table and network_offering_details table ?

**Comment by DensonNgumo** [comment]
> @DensonNgumo I see all offerings in both cases. and lso with or without the guestiptype=L2.
> 
> Did you create any offerings yourself? Did you create this zone (and enable it) before or after creating any offerings?
> 
> You say you see this in 4.18.0 and in 4.19.0.
> 
> Did you upgrade and tested again, or create a new installation?
> 
> Can you create a new offering, with the same parameters and see if this displays?

I created the one labeled 'SIP_Network' to test if the issue was the default offerings. The issue was there without the custom offering.

**Comment by DensonNgumo** [comment]
> @DensonNgumo I see all offerings in both cases. and lso with or without the guestiptype=L2.
> 
> Did you create any offerings yourself? Did you create this zone (and enable it) before or after creating any offerings?
> 
> You say you see this in 4.18.0 and in 4.19.0.
> 
> Did you upgrade and tested again, or create a new installation?
> 
> Can you create a new offering, with the same parameters and see if this displays?

The issue was first observed in version 4.18.0. I then upgraded to 4.19.0 to see if it would resolve but there was no change. The Zone had been created during the initial setup. The zone was up and enabled before any custom offerings. The custom offering was created just to troubleshoot the issue.

**Comment by weizhouapache** [comment]
> SIP_Network

@DensonNgumo can you remove this offering if it is not in use ?
can you update the access of offering "DefaultL2NetworkOfferingConfigDriveVlan" to "Public" to "All zones" ?

**Comment by DaanHoogland** [comment]
yes, so in case you don't know; go through all the offerings that should be visible, press this button: ![image](https://github.com/apache/cloudstack/assets/2486961/961cf47f-0875-4c27-8450-7b73663c187d), and update the access settings in the dialogs.

**Comment by DensonNgumo** [comment]
I have removed the custom offering. All the default offerings are set to "Public" access. Changing the zone to "All zones" leaves the field as blank.
![offering](https://github.com/apache/cloudstack/assets/11226482/21084038-81aa-424e-a041-9da53e1af8ec)
![all-offerings](https://github.com/apache/cloudstack/assets/11226482/5f124692-c682-456d-aa3c-2ba73401f43d)
Still not able to select a network offering.

**Comment by DaanHoogland** [comment]
ok, this is not as intended but what if you set the zone specifically for one of the offerings, do you then see it?

**Comment by DensonNgumo** [comment]
https://github.com/apache/cloudstack/issues/9193#issuecomment-2160315660
That's how it was. The offering I wanted to use had the zone specified.
None of the offerings has a zoneid when pulling the info using cmk
![zone-offerings](https://github.com/apache/cloudstack/assets/11226482/47ad1f77-1d67-4ed6-b74e-9611e915f3ca)

**Comment by weizhouapache** [comment]
> https://github.com/apache/cloudstack/issues/9193#issuecomment-2160315660
> That's how it was. The offering I wanted to use had the zone specified.
> None of the offerings has a zoneid when pulling the info using cmk
> ![zone-offerings](https://github.com/apache/cloudstack/assets/11226482/47ad1f77-1d67-4ed6-b74e-9611e915f3ca)
> 

That is strange.
I was not able to reproduce the issue. Everything looks good in my testing.
I believe @DaanHoogland is not able to reproduce either

Can you share the mgmt server log when list network offerings with zoneid?

**Comment by DensonNgumo** [comment]
[cs_mgt_log_14_5_24.txt](https://github.com/user-attachments/files/15834683/cs_mgt_log_14_5_24.txt)
Find the log attached. Selecting the drop down does not actually prompt any recording on the logs. It is like the drop down is not triggering the API query.

**Comment by DaanHoogland** [comment]
@DensonNgumo , I would not expect the opeing of the dropdown to trigger the API call, but before that: the opening of the dialog.

**Comment by weizhouapache** [comment]
@DensonNgumo 
I expected the log gives more clue. but it does not.
the issue is because of the empty result when you list network offerings with zoneid
https://github.com/apache/cloudstack/issues/9193#issuecomment-2164671480

I am not able to reproduce the issue ..

**Comment by DaanHoogland** [comment]
@DensonNgumo This seems to be a UI issue in 4.19 that is fixed in the HEAD of 4.19 branch and in main. I have not found the matching PR.

**Comment by DensonNgumo** [comment]
Is there anything I can do to get around it? As it is, I am unable to create any network.

**Comment by DaanHoogland** [comment]
@DensonNgumo , use the API, is all I can think of now.

**Comment by UAnton** [comment]
Get the same problem after upgrading to 4.19.0.2

**Comment by yadvr** [comment]
@DensonNgumo @UAnton can you try to upgrade to 4.19.1.0 and also suggest following checklist:

- Do you have the zone enabled and specific network service providers also enabled
- Can you share screenshot from your zone's physical networks' network service providers tab
- Can you try this in a private browser tab/window?
- Do you have available/free VLANs in that zone

**Comment by GeorgeLisandru** [comment]
HI @rohityadavcloud! 
We are facing the same problem with a fresh deployment of ACS 4.19.1.0. We have our system VMs up and running and we are able to migrate them on any of the 3 nodes that we are currently using.
But, after we deploy the Zone (Advanced), without making any changes, we don’t have any network offering for Isolated/L2/Shared. As you can imagine, being a fresh deployment, we do have available VLANs, I try accessing from different browsers but the outcome is always the same.
Is there something that we are missing ?

**Comment by borisstoyanov** [comment]
@DensonNgumo @GeorgeLisandru @UAnton 
I've tried reproducing the issue and the only way I've managed to reproduce it when I've disabled the 'Network Service Providers' on the physical network in the newly added zone settings. In particular the VirtualRouter and VpcVirtualRouter, can you please double check that you have those enabled? 

That is Zone - Select zone -> Physical network -> select PN -> Network Service Providers 
![Screenshot 2024-07-29 at 15 11 39](https://github.com/user-attachments/assets/8039f8c7-f984-44a6-953d-a60f14e17b74)

Only when I have VR and VPC VR disabled I get blank for selecting offering at this zone 

![Screenshot 2024-07-29 at 15 12 24](https://github.com/user-attachments/assets/1517e7f9-7163-48ad-8b0f-4791d9c19dda)


![Screenshot 2024-07-29 at 15 12 51](https://github.com/user-attachments/assets/54d00102-e6c3-4503-99f8-67e907788903)

Please let me know if this is the case

**Comment by yadvr** [comment]
To add to what @borisstoyanov has shared, this usually can happen if the zone deployment form has errors or for some reason the zone deployment didn't go as expected (for example silent errors) and only then offerings are not listed as the zone doesn't have the right supported network service providers enabled/configured.

@GeorgeLisandru can you go to the UI -> Infrastructure -> Zone -> select your zone -> go the physical network and select that -> click on the network service providers (NSP) tab; and share screenshot of this page. Do you see the VirtualRouter and VPCVirtualRouter network service providers in red (i.e. disabled)? If they are indeed disabled, can you try to enable them and see if you can then see the network offerings listed?

**Comment by GeorgeLisandru** [comment]
Thank you very much for your answers!


Indeed this is the case. I have VirtualRouter, VPCVirtualRouter and InternalLbVm in stop state and I am not able to start any of them. Below is a screenshot of the error message that I get.

![image](https://github.com/user-attachments/assets/ab738d52-5645-4c9e-ab4c-fc650d31169e)

**Comment by DensonNgumo** [comment]
> @DensonNgumo @UAnton can you try to upgrade to 4.19.1.0 and also suggest following checklist:
> 
> * Do you have the zone enabled and specific network service providers also enabled
> * Can you share screenshot from your zone's physical networks' network service providers tab
> * Can you try this in a private browser tab/window?
> * Do you have available/free VLANs in that zone

Hi,
Yes, the zone was enabled. On initial setup everything worked as expected and I never got any errors during installation, had even created my first shared network. The network service providers are enabled as shown below:
![cs-physical_network](https://github.com/user-attachments/assets/d86470fa-9fef-429a-bf6d-f4db616e640e)
Once I observed the issue is when I upgraded from 4.18 to 4.19 but the issue remained.

**Comment by borisstoyanov** [comment]
I think we're looking at 2 separate issues. 

@DensonNgumo could you please confirm the following: 
1. current version to the dot
2. you can still reproduce the issue
3. please do an api call with cmk, trying to create a network with the offering you're not able to see, and send the management logs

@GeorgeLisandru in your case I believe you might need to: 
1. execute the following with cmk: configure virtualrouterelement enabled=true id= [id of the VP provider] 
2. if not successful please provide the logs.

**Comment by DensonNgumo** [comment]
> I think we're looking at 2 separate issues.
> 
> @DensonNgumo could you please confirm the following:
> 
> 1. current version to the dot
> 2. you can still reproduce the issue
> 3. please do an api call with cmk, trying to create a network with the offering you're not able to see, and send the management logs
> 
> @GeorgeLisandru in your case I believe you might need to:
> 
> 1. execute the following with cmk: configure virtualrouterelement enabled=true id= [id of the VP provider]
> 2. if not successful please provide the logs.

 My current version is 4.19.0.1. Issue is still present. I will attempt the same with cmk and share the logs.

**Comment by DensonNgumo** [comment]
> I think we're looking at 2 separate issues.
> 
> @DensonNgumo could you please confirm the following:
> 
> 1. current version to the dot
> 2. you can still reproduce the issue
> 3. please do an api call with cmk, trying to create a network with the offering you're not able to see, and send the management logs
> 
> @GeorgeLisandru in your case I believe you might need to:
> 
> 1. execute the following with cmk: configure virtualrouterelement enabled=true id= [id of the VP provider]
> 2. if not successful please provide the logs.

Creating the network using api calls via cmk is successful. 
![cmk-network](https://github.com/user-attachments/assets/1ed4c909-fbc0-43b8-b398-eaf56bf0b2ac)

**Comment by DensonNgumo** [comment]
> > I think we're looking at 2 separate issues.
> > @DensonNgumo could you please confirm the following:
> > 
> > 1. current version to the dot
> > 2. you can still reproduce the issue
> > 3. please do an api call with cmk, trying to create a network with the offering you're not able to see, and send the management logs
> > 
> > @GeorgeLisandru in your case I believe you might need to:
> > 
> > 1. execute the following with cmk: configure virtualrouterelement enabled=true id= [id of the VP provider]
> > 2. if not successful please provide the logs.
> 
> Creating the network using api calls via cmk is successful. ![cmk-network](https://private-user-images.githubusercontent.com/11226482/365157392-1ed4c909-fbc0-43b8-b398-eaf56bf0b2ac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2MjQ4NjAsIm5iZiI6MTcyNTYyNDU2MCwicGF0aCI6Ii8xMTIyNjQ4Mi8zNjUxNTczOTItMWVkNGM5MDktZmJjMC00M2I4LWIzOTgtZWFmNTZiZjBiMmFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA2VDEyMDkyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMwMGYyODA0NWQ4NGNmYWRmNjgxNTcwN2UzNzA3ZjFiMzk3NjYyZTk0OWY0ODBlODdmZDg2NWJjYjI2MDcxZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ITcT4qhuKJe519G2etwLrT9z97dhTXrRBYOs5XqAneg)

I upgraded to the latest version 4.19.1.1 and the issue of not being able to select the network offering via the UI remains.

**Comment by int0x21** [comment, marked as answer]
This happens when you have a tag on the network, remove the tag and it works. Spent all day on this 
Fresh install.

**Reply by weizhouapache** [reply]
@sidescale 
which ACS version do you use ?

**Comment by DensonNgumo** [comment]
> This happens when you have a tag on the network, remove the tag and it works. Spent all day on this Fresh install.

Hi, please elaborate. Tag on which section exactly? I have not set any network tags on the offerings.

**Reply by samapraku** [reply]
@DensonNgumo tag on the physical network. I also had the same issue and removing the tag as mentioned by @int0x21  fixed it for me.

I think it is possibly related to this issue  https://github.com/apache/cloudstack/issues/6747 I opened back in 2022 regarding the same problem.  The issue was closed but apparently the problem is still there

**Reply by int0x21** [reply]
When creating the zone and setting up network you get the option off setting tags on the physical network, that is what caused all the issues for me. I really like the potential of cloudstack but some of these things needs to be  cleaned up. 
@DensonNgumo I'm glad my suggestion solved your issue :)

**Reply by DensonNgumo** [reply]
I do confirm that removing the tag on the physical network did indeed resolve the issue. Thanks @int0x21

**Comment by rajujith** [comment]
@int0x21  hopefully this will be fixed in 4.19.2: https://github.com/apache/cloudstack/issues/9673

**Comment by UAnton** [comment]
Hi, 
This happens without tags. I again get this problem on 4.20.

**Comment by Pearl1594** [comment]
When you say that there are no tags @UAnton , do you mean, on the Physical network or the network offering(s)?
Because, for no tags on Physical network - then atleast the default offerings should get listed
(Side note: If the network offerings are tagged, they need to have a tag that matches the physical network)

**Reply by UAnton** [reply]
<img width="670" alt="Screenshot 2025-02-11 at 20 57 58" src="https://github.com/user-attachments/assets/1640ab88-3774-4680-bf26-115ba1222c7f" />
<img width="2246" alt="Screenshot 2025-02-11 at 20 58 35" src="https://github.com/user-attachments/assets/0894f295-9834-4833-b457-c9deb72675aa" />

1. When I add all Traffic types to the one Physical Network - no Network Offerings
2. When I try to add 2 Physical Networks and separate MGMT traffic from Public and Guest - SystemVMs doesn't start. 
Every time I get the problems with CS network model.

**Comment by UAnton** [comment]
<img width="1178" alt="Screenshot 2025-02-11 at 21 51 59" src="https://github.com/user-attachments/assets/1ae1d0d2-775e-46cf-be00-4508c2a3b2a1" />
<img width="1577" alt="Screenshot 2025-02-11 at 21 52 30" src="https://github.com/user-attachments/assets/7230c714-03fa-47f1-b524-f7e8ee55843f" />

I was trying different variants, but every type has 2 problems. 1 - No network offerings. 2 - SystemVMs don't starting.
I read documentation, forums, and GitHub issues but nothing helped. Watch a video from ShapeBlue about Cloudstack and XCP-NG. Repeat Step-by-Step from the video but every time get this problem!

**Comment by weizhouapache** [comment]
thanks @sidescale 

@rajujith has already created a doc PR
https://github.com/apache/cloudstack-documentation/pull/481
